src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -614,8 +614,20 @@ public bool IsHardforkEnabled(Hardfork hardfork)
         {
             if (PersistingBlock is null)
                 return true;
+
             if (!ProtocolSettings.Hardforks.TryGetValue(hardfork, out uint height))
-                return true;
+                return false;
```

Comment:
This change solves my problem. However, note that default behaviour is the empty set of hardforks: https://github.com/neo-project/neo/blob/946410748ba49a3321f2e6fbccb4adcc8c5f1938/src/Neo/ProtocolSettings.cs#L164

Since we'd like to enable all hardforks started from zero height if none of them specified in the config, then here we have to add some code checking that the list of hardforks isn't empty (and return `true` if it's empty).

Reply:
wait a minute, you wish to enable all hardfork if they are not configed?

Reply:
If none of the hardforks are configured, then yes, all of them are treated as "enabled" by default. If at least one hardfork is configured, then the rest of them should be treated as "disabled". That's the way how it works in NeoGo, but it can be discussed. @roman-khimov

Reply:
hardfork being considered as enabled without being configed.,,when when do we enabled them？ consider them as enabled since the genesis?

Reply:
> when when do we enabled them？ consider them as enabled since the genesis?

If none of the hardforks are configured in the node config, then yes, enable them since genesis. And BTW, that's the behaviour NeoC# and NeoGo had previously.

Reply:
> And BTW, that's the behaviour NeoC# and NeoGo had previously.

I thought you wish to change that logic

Reply:
> I thought you wish to change that logic

I wish to change another logic. The logic which I wish to change is that if at least one hardfork is specified in the node config, then the rest of them (the ones that missing from the config) should be treated as disabled. But currently in this case  NeoC# node treats the missing hardfork as enabled.

Reply:
See the desired way: https://github.com/nspcc-dev/neo-go/blob/11c0f13d066b27c04cc1d46058928e32db9bddf6/pkg/core/interop/context.go#L403-L409:
```
// IsHardforkEnabled tells whether specified hard-fork enabled at the current context height.
func (ic *Context) IsHardforkEnabled(hf config.Hardfork) bool {
	height, ok := ic.Hardforks[hf.String()]
	if ok {
		return ic.BlockHeight() >= height
	}
	return len(ic.Hardforks) == 0 // Enable each hard-fork by default.
}
```

Reply:
Agree with @AnnaShaleva no hardforks should be all enabled by default